File: app/src/main/java/com/hrithikvish/cbsm/utils/SharedPrefManager.java
package com.hrithikvish.cbsm.utils;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;

public class SharedPrefManager {
    private final SharedPreferences sharedPreferences;
    public SharedPrefManager(Context context) {
        sharedPreferences = context.getSharedPreferences(Constants.SHARED_PREFERENCE_KEY, Context.MODE_PRIVATE);
    }

    public void putBoolean(String key, Boolean value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public void putObject(String key, Object object) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        Gson gson = new Gson();
        String objectString = gson.toJson(object);
        editor.putString(key, objectString);
        editor.apply();
    }

    public String getObject(String key) {
        return sharedPreferences.getString(key, null);
    }

    public Boolean getBoolean(String key) {
        return sharedPreferences.getBoolean(key, false);
    }

    public void putString(String key, String value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public String getString(String key) {
        return sharedPreferences.getString(key, null);
    }

    public void clear() {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.clear();
        editor.apply();
    }

}

File: app/src/main/java/com/hrithikvish/cbsm/utils/SharedPrefManager.kt
package com.hrithikvish.cbsm.utils

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson

class SharedPrefManager(context: Context) {
    private val sharedPreferences: SharedPreferences = context.getSharedPreferences(
        Constants.SHARED_PREFERENCE_KEY,
        Context.MODE_PRIVATE
    )

    fun putBoolean(key: String?, value: Boolean) {
        val editor = sharedPreferences.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun putObject(key: String?, `object`: Any?) {
        val editor = sharedPreferences.edit()
        val gson = Gson()
        val objectString = gson.toJson(`object`)
        editor.putString(key, objectString)
        editor.apply()
    }

    fun getObject(key: String?): String? {
        return sharedPreferences.getString(key, null)
    }

    fun getBoolean(key: String?): Boolean {
        return sharedPreferences.getBoolean(key, false)
    }

    fun putString(key: String?, value: String?) {
        val editor = sharedPreferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun putInt(key: String?, value: Int) {
        val editor = sharedPreferences.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    fun getInt(key: String?): Int {
        return sharedPreferences.getInt(key, -1)
    }

    fun getString(key: String?): String? {
        return sharedPreferences.getString(key, null)
    }

    fun clear() {
        val editor = sharedPreferences.edit()
        editor.clear()
        editor.apply()
    }
}